File: ROS/ros_learn/src/comm_msg_custom/scripts/topic/talker.py
#! /usr/bin/env python

import rospy
from comm_msg_custom.msg import person

if __name__ == "__main__":
    rospy.init_node("talker_person_p")
    pub = rospy.Publisher("chat_test",person,queue_size=10)
    p = person()
    p.name = "catcolia"
    p.age = 500
    p.height = 1.80

    r = rospy.Rate(1)
    while not rospy.is_shutdown():
        pub.publish(p)
        p.age += 1
        r.sleep()
        rospy.loginfo("Send message, Name:%s Age:%d Height:%.2f",p.name,p.age,p.height)

File: ROS/ros_learn/src/comm_msg_custom/src/topic/talker.cpp
#include "ros/ros.h"
#include "comm_msg_custom/person.h"

int main(int argc, char* argv[])
{

  ROS_INFO("This is a talker node");
  setlocale(LC_ALL, "");

  ros::init(argc, argv, "topic_conn_talker");
  ros::NodeHandle nh;
  ros::Publisher publisher =
      nh.advertise<comm_msg_custom::person>("chat_test", 10);

  comm_msg_custom::person person;
  person.age = 50;
  person.height = 180.1;
  person.name = "Catcolia";

  ros::Rate r(1);

  while (ros::ok()) {

    publisher.publish(person);
    person.age += 1;

    ROS_INFO("Publish message.  Name:%s Age:%d Height:%f", person.name.c_str(),
             person.age, person.height);

    r.sleep();
    ros::spinOnce();
  }

  return 0;
}
